### PR TITLE
Muhajir goal deallocation

### PR DESCRIPTION
**Pull request by M-Sayer**

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -35,26 +44,107 @@ const convertTransactionsToDollars = (arr, type) => {
 
 const getDifference = (oldAmt, newAmt) => {
   return (oldAmt - newAmt) * -1;
```
Comment (mischkiey) [inline src/helpers.js:46]: Alternatively:

`return -(oldAmt - newAmt)`

**Comment by jenthura** [comment]
Migrations are working great!
Transactions router is using the old /create for the POST
Also, there's no goals router at all. I must be missing something.
Fraction.js? New packages! :D

File: src/helpers.js [hunk 2]
```diff
@@ -35,26 +44,107 @@ const convertTransactionsToDollars = (arr, type) => {
...
+  ratios.forEach(ratio => {
+    const share = Math.floor(amount * ratio/total);
+    results.push(share);
+    remainder -= share;
+  });
+
+  //add one from remainder to each share until no remainder
+  for (let i = 0; remainder > 0; i++) {
+    results[i] = results[i] + 1;
+    remainder --;
```
Comment (mischkiey) [inline src/helpers.js:75]: Interesting. Conceptually, there shouldn't be any remainder but there is, as a result of the accumulated increments from using Math.floor.

**Review by mischkiey — APPROVED** [review]
Covers most (if not all) edge cases. Super cool what you did here

**Pull request merged**